WeatherApp/DAL/WeatherRepository.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace DAL
+{
+    public class WeatherRepository : IWeatherRepository
+    {
+        private readonly string _apiKey;
+        private readonly HttpClient _httpClient = new HttpClient();
+
+        public WeatherRepository(string apiKey)
+        {
+            _apiKey = apiKey;
+        }
+
+        public async Task<double> GetTemperatureByCityNameAsync(string cityName)
+        {
+            using HttpResponseMessage response = await _httpClient.GetAsync($"https://api.openweathermap.org/data/2.5/weather?q={cityName}&units=metric&APPID={_apiKey}");
+            response.EnsureSuccessStatusCode();
+
+            string responseBody = await response.Content.ReadAsStringAsync();
+            Models.OpenWeather.WeatherResponse weather = JsonSerializer.Deserialize<Models.OpenWeather.WeatherResponse>(responseBody);
```

Comment:
Please add using Models.OpenWeather; to prevent long type names

Reply:
I added "using" to prevent long type names.